### PR TITLE
feat(Toaster): Return `close` function for individual toasts

### LINKED ISSUE #365: programmatically close toasts
#### Feature

Can `toaster[toastMethod]` return a function reference that can be used to close the toast?

We have a "quick export" feature in our application, and we're currently using toasts as a simple way to give the users
feeedback on it, the application is very simple and not really meant to be used while an export is running, which can take a few minutes
depending on several factors. It would be nice if we could programmatically close a toast instead of querying the DOM
for the close button and simulating the user clicking it.

We'd like to close the "in-progress" toast just as a "success" toast
comes up, right now the user has to manually close the toast, and both exist at the same time even though the original toast is no
longer relevant.

image

#### Examples

```typescript
const { close } = toaster.informational(<><ProgressSpinner />Your Export is running...</>, { type: "persisting", hasCloseButton: true});
await runExport();
close().
```

Also, we used iTwin-UI to rewrite our selection screen in two days and make it 10,000x better. Amazing work on this
package, it's never been so easy to make beautiful, consistent designs

### PR DESCRIPTION
**Pull request by gretanausedaite**
Toaster create functions now returns an object with Close function which can be used to programmatically close toasts.

Closes #365

## Checklist

- [x] Add meaningful unit tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
- [x] Add component features demo in Storybook (different stories)
- [x] Approve test images for new stories
- [x] Add screenshots of the key elements of the component

### CODE REVIEW

File: src/core/Toast/Toaster.tsx [hunk 1]
```diff
@@ -115,6 +116,23 @@ export default class Toaster {
...
+    this.toasts = this.toasts.map((toast) => {
+      if (toast.id === toastId) {
+        return {
+          ...toast,
+          isVisible: false,
+        };
+      }
+      return {
+        ...toast,
+      };
```
Comment (mayank99) [inline src/core/Toast/Toaster.tsx:131]:
```suggestion
      return {
        ...toast,
        isVisible: toast.id !== toastId,
      };
```

File: src/core/Toast/Toaster.tsx [hunk 2]
```diff
@@ -115,6 +116,23 @@ export default class Toaster {
     );
   }
 
+  // close toast function
+  closeToast(toastId: number): void {
+    console.log('calling close toast');
```
Comment (mayank99) [inline src/core/Toast/Toaster.tsx:121]: Remove

File: src/core/Toast/Toaster.tsx [hunk 3]
```diff
@@ -115,6 +116,23 @@ export default class Toaster {
     );
   }
 
+  // close toast function
```
Comment (mayank99) [inline src/core/Toast/Toaster.tsx:119]: This comment is redundant

File: src/core/Toast/Toaster.tsx [hunk 4]
```diff
@@ -115,6 +118,23 @@ export default class Toaster {
...
+  closeToast(toastId: number): void {
```
[the remark this answers is not in the document]
Comment (mayank99) [reply src/core/Toast/Toaster.tsx:120]: Might want to make it private

File: stories/core/Toasts.stories.tsx [hunk 5]
```diff
@@ -343,3 +343,66 @@ export const AnchorToButton: Story<ToastProps & ToasterSettings> = ({
...
+    setTimeout(() => {
+      close();
+      toaster.positive('Process completed', {
+        duration,
+        hasCloseButton,
+        link,
+        type: 'temporary',
+        onRemove,
+      });
+    }, 7000);
```
Comment (bentleyvk) [inline stories/core/Toasts.stories.tsx:394]: 7s might be too much 😀

File: src/core/Toast/Toaster.test.tsx [hunk 6]
```diff
@@ -24,11 +24,12 @@ function assertAddedToast(
   toast: ToastProps,
   category: ToastCategory,
   content: string,
+  options: ToastOptions,
```
[the remark this answers is not in the document]
Comment (gretanausedaite) [reply src/core/Toast/Toaster.test.tsx]: Made this prop optional in both added and deleted functions.

**Review by bentleyvk — APPROVED** [review]

**Review by mayank99 — APPROVED** [review]

**Pull request merged**